tests/php/Control/LoginSessionControllerTest.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace SilverStripe\SessionManager\Tests\Control;
+
+use HttpRequest;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\Middleware\ConfirmationMiddleware\Url;
+use SilverStripe\Control\Session;
+use SilverStripe\Control\Tests\HttpRequestMockBuilder;
+use SilverStripe\Dev\FunctionalTest;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+use SilverStripe\Security\SecurityToken;
+use SilverStripe\SessionManager\Control\LoginSessionMiddleware;
+use SilverStripe\SessionManager\Models\LoginSession;
+
+class LoginSessionControllerTest extends FunctionalTest
+{
+    protected static $fixture_file = '../LoginSessionTest.yml';
+
+    use HttpRequestMockBuilder;
+
+    /** @var Session */
+    private $session;
+
+    /** @var HttpRequest */
+    private $request;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $sessionID = $this->idFromFixture(LoginSession::class, 'x1');
+        $this->session = new Session(['activeLoginSession' => $sessionID]);
+        $this->request =  $this->buildRequestMock('/', [], [], null, $this->session);
+        $this->request->method('getIP')->willReturn('192.168.0.1');
+
+        $memberID = $this->idFromFixture(Member::class, 'owner');
+        $this->logInAs($memberID);
+        SecurityToken::enable();
+        $this->session->set('SecurityID', SecurityToken::inst()->getValue());
+    }
+
+    public function testRemove()
+    {
+        $sessionID = $this->idFromFixture(LoginSession::class, 'x3');
+        $token = SecurityToken::inst()->getValue();
+        $response = Director::test(
+            "loginsession/$sessionID?SecurityID=$token",
+            null,
+            $this->session,
+            'DELETE'
+        );
+
+        $this->assertEquals(
+            '200',
+            $response->getStatusCode(),
+            'User can invalidate their session with a valid CSRF token'
+        );
+        $body = json_decode($response->getBody(), true);
+        $this->assertEquals(['message' => 'Successfully logged out of device.'], $body);
```

Comment:
I'm a bit ambivalent about asserting the exact message being returned for all these scenarios. I'm thinking maybe asserting the response code is good enough.

This will break the test if the user has set up their site to a non-English language ... while not terrible, it's not great either.